src/Storages/DataLakes/Iceberg/StorageIceberg.cpp

```diff
@@ -52,8 +73,12 @@ void StorageIceberg::updateConfigurationImpl(ContextPtr local_context)
 {
     const bool updated = base_configuration.update(local_context);
     auto new_metadata = parseIcebergMetadata(base_configuration, local_context);
+
+    if (!current_metadata)
+        current_metadata = parseIcebergMetadata(base_configuration, local_context);
```

Comment:
If we don't have `current_metadata`, we will do the same work twice here. We can check if we don't have `current_metadata` just parse it and update the configuration, because we don't have 2 different metadatas to compare